torch/csrc/jit/passes/xnnpack_rewrite.cpp

```diff
@@ -0,0 +1,108 @@
+#include <torch/csrc/jit/passes/constant_pooling.h>
+#include <torch/csrc/jit/passes/constant_propagation.h>
+#include <torch/csrc/jit/passes/xnnpack_rewrite.h>
+#include <torch/csrc/jit/passes/fuse_linear.h>
+#include <torch/csrc/jit/passes/graph_rewrite_helper.h>
+#include <torch/csrc/jit/passes/subgraph_rewrite.h>
+#include <torch/csrc/jit/ir/subgraph_matcher.h>
+
+namespace torch {
+namespace jit {
+
+#ifdef USE_XNNPACK
+
+namespace {
+
+void insertXNNPACKLinearOp(std::shared_ptr<Graph>& graph) {
+  std::string linear_before_inline = R"(
+    graph(%linear, %input, %weight, %bias):
+        %r = prim::CallFunction(%linear, %input, %weight, %bias)
```

Comment:
This one makes me a bit nervous.  Will this match any 4-arg function called "linear", even if it's not aten::linear or torch.nn.functional.linear?

Reply:
This is a good point. Your observation I think is correct. Let me think about this as to what are the caveats here. Tagging @jerryzh168, to clarify something I might have missed.

Reply:
So this is doable. I think we can do more stringent checking. I will do it outside of this stack, since there are a few common functions we share with quantization and if we make a change there, it affects both. And I will have to make changes in quantization.cpp as well.